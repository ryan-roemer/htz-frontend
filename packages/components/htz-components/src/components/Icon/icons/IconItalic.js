/* *************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change this Icon component , it is generated
 * from the `iconTamplate.js` file the parent directory.
 * *************************************************************** */
import React from 'react';
import { createComponent, } from 'react-fela';
import iconStyle from '../iconStyle';
import { iconPropTypes, iconDefaultProps, } from '../iconPropTypes';
import { attrsPropType, } from '../../../propTypes/attrsPropType';

const IconItalic = createComponent(
  iconStyle,
  UnstyledIconItalic,
  [ 'attrs', 'onClick', ]
);

IconItalic.propTypes = iconPropTypes;
IconItalic.defaultProps = iconDefaultProps;


// Underlying component
UnstyledIconItalic.propTypes = {
  attrs: attrsPropType,
};

UnstyledIconItalic.defaultProps = {
  attrs: null,
};

function UnstyledIconItalic({ attrs, ...props }) {
  return (
    <svg width="1em" height="1em" viewBox="0 0 256 256" {...props} {...attrs} >  <path fill="currentColor" d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z" /></svg>
  );
}


export default IconItalic;
