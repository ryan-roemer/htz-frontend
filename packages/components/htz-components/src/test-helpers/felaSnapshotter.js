import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
