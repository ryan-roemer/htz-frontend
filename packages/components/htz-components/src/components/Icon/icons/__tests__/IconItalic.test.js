
/* global window */
/* *************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change the styleguide example, it is generated
 * from the `embedsFileTemplate.js` file is this directory.
 * *************************************************************** */
import React from 'react';
import felaSnapshotter from '../../../../test-helpers/felaSnapshotter';
import IconItalic from '../IconItalic';

describe('<IconItalic />', () => {
  /* The error tests are left as documentation, but are commented out for the redability of the test results */
  // describe('Errors', () => {
  //   it('throw an error when the value passed to the "size" prop isn't a number', () => {
  //     expect(() => felaSnapshotter(<IconItalic size="2rem" />)).toThrow(
  //       /^An Icon's "size" prop may only be passed a "number"/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconItalic color="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconItalic color={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconItalic fill="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconItalic fill={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  // });
  describe('DOM element', () => {
    testCase('render correctly when no props are passed', <IconItalic />);
    testCase(
      'pass attributes defined in the "attrs" prop to the DOM element',
      <IconItalic attrs={{ 'aria-hidden': true, }} />
    );
    testCase('add "id" attribute to DOM element', <IconItalic id="hasId" />);
    testCase(
      'add "onClick" attribute to DOM element',
      <IconItalic id={e => console.log('a')} />
    );
  });

  describe('styles', () => {
    testCase("set icon's color", <IconItalic color={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's color responsively",
      <IconItalic
        color={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's fill", <IconItalic fill={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's fill responsively",
      <IconItalic
        fill={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's size", <IconItalic size={6} />);
    testCase(
      "set icon's size responsively",
      <IconItalic
        size={[ { until: 'xl', value: 6, }, { from: 'xl', value: 7, }, ]}
      />
    );
    testCase(
      'set misc styles on an icon',
      <IconItalic miscStyles={{ position: 'relative', }} />
    );
    testCase(
      'responsively set misc styles on an icon',
      <IconItalic
        miscStyles={{
          position: [
            { until: 'xl', value: 'relative', },
            { from: 'xl', value: 'static', },
          ],
        }}
      />
    );
  });
});

function testCase(name, Component, { only = false, } = {}) {
  const testFunc = only ? it.only : it;
  testFunc(name, () => {
    const { component, styles, } = felaSnapshotter(Component);
    expect(component).toMatchSnapshot();
    expect(styles).toMatchSnapshot();
  });
}
