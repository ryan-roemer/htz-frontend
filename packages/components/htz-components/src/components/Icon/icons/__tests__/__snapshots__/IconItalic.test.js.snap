// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconItalic /> DOM element add "id" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id="hasId"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> DOM element add "id" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconItalic /> DOM element add "onClick" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id={[Function]}
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> DOM element add "onClick" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconItalic /> DOM element pass attributes defined in the "attrs" prop to the DOM element 1`] = `
<div>
  <svg
    aria-hidden={true}
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> DOM element pass attributes defined in the "attrs" prop to the DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconItalic /> DOM element render correctly when no props are passed 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> DOM element render correctly when no props are passed 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconItalic /> styles responsively set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles responsively set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{position:relative}}@media (min-width: 80em){.htz-e{position:static}}"`;

exports[`<IconItalic /> styles set icon's color 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles set icon's color 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:#0B7EB5}.htz-c{fill:#FFF}"`;

exports[`<IconItalic /> styles set icon's color responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles set icon's color responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-d{fill:#FFF}@media (max-width: 79.9375em){.htz-b{color:#0B7EB5}}@media (min-width: 80em){.htz-c{color:#2D2D2D}}"`;

exports[`<IconItalic /> styles set icon's fill 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles set icon's fill 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#0B7EB5}"`;

exports[`<IconItalic /> styles set icon's fill responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles set icon's fill responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}@media (max-width: 79.9375em){.htz-c{fill:#0B7EB5}}@media (min-width: 80em){.htz-d{fill:#2D2D2D}}"`;

exports[`<IconItalic /> styles set icon's size 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles set icon's size 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{font-size:6rem}"`;

exports[`<IconItalic /> styles set icon's size responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles set icon's size responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{font-size:6rem}}@media (min-width: 80em){.htz-e{font-size:7rem}}"`;

exports[`<IconItalic /> styles set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M147.7 229.7l-2.2 7H53.6l2.2-7h5c6.6 0 12.2-1.2 16.8-3.6 4.6-2.4 7.7-5.2 9.4-8.3 1.7-3.2 4.1-10.1 7.3-20.8l41.9-138.4c3.3-10.7 5-17.4 5-20 0-7.7-6.9-11.5-20.7-11.5l-5.8-1 2.2-7h85.6l-2.2 7h-4.5c-5.1 0-9.8.9-14 2.7-4.2 1.8-7.1 4.3-8.9 7.5-1.8 3.2-4.7 11.5-8.7 24.9l-40.6 133.3c-3.1 10.2-4.7 16.6-4.7 19.2 0 5.2 2.3 9.1 6.8 11.5 4.5 2.4 9.8 3.6 15.6 3.6l6.4.9z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconItalic /> styles set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{position:relative}"`;
